File: app/src/main/java/com/niceplaces/niceplaces/activities/SettingsActivity.java
package com.niceplaces.niceplaces.activities;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.CheckBox;
import android.widget.EditText;

import com.niceplaces.niceplaces.R;
import com.niceplaces.niceplaces.controllers.PrefsController;

public class SettingsActivity extends AppCompatActivity {

    private EditText mEditTextRadius, mEditTextRefreshTime, mEditTextRefreshDistance;
    private PrefsController mPrefs;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);
        getSupportActionBar().setTitle(R.string.settings);
        mEditTextRadius = findViewById(R.id.edittext_distance_radius);
        mEditTextRefreshTime = findViewById(R.id.edittext_position_refresh_time);
        mEditTextRefreshDistance = findViewById(R.id.edittext_position_refresh_distance);
        mPrefs = new PrefsController(this);
        mEditTextRadius.setText(String.valueOf(mPrefs.getDistanceRadius()));
        mEditTextRefreshTime.setText(String.valueOf(mPrefs.getLocationRefreshTime()));
        mEditTextRefreshDistance.setText(String.valueOf(mPrefs.getLocationRefreshDistance()));
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        mPrefs.setDistanceRadius(Float.parseFloat(mEditTextRadius.getText().toString()));
        mPrefs.setLocationRefreshTime(Integer.parseInt(mEditTextRefreshTime.getText().toString()));
        mPrefs.setLocationRefreshDistance(Integer.parseInt(mEditTextRefreshDistance.getText().toString()));
    }
}

File: app/src/main/java/com/niceplaces/niceplaces/activities/SettingsActivity.kt
package com.niceplaces.niceplaces.activities

import android.os.Bundle
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import com.niceplaces.niceplaces.R
import com.niceplaces.niceplaces.controllers.PrefsController

class SettingsActivity : AppCompatActivity() {
    private var mEditTextRadius: EditText? = null
    private var mEditTextRefreshTime: EditText? = null
    private var mEditTextRefreshDistance: EditText? = null
    private var mPrefs: PrefsController? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)
        supportActionBar!!.setTitle(R.string.settings)
        mEditTextRadius = findViewById(R.id.edittext_distance_radius)
        mEditTextRefreshTime = findViewById(R.id.edittext_position_refresh_time)
        mEditTextRefreshDistance = findViewById(R.id.edittext_position_refresh_distance)
        mPrefs = PrefsController(this)
        mEditTextRadius?.setText(mPrefs!!.distanceRadius.toString())
        mEditTextRefreshTime?.setText(mPrefs!!.locationRefreshTime.toString())
        mEditTextRefreshDistance?.setText(mPrefs!!.locationRefreshDistance.toString())
    }

    override fun onBackPressed() {
        super.onBackPressed()
        mPrefs!!.distanceRadius = mEditTextRadius!!.text.toString().toFloat()
        mPrefs!!.locationRefreshTime = mEditTextRefreshTime!!.text.toString().toInt()
        mPrefs!!.locationRefreshDistance = mEditTextRefreshDistance!!.text.toString().toInt()
    }
}